test/functional/wallet_basic.py

```diff
@@ -496,6 +496,26 @@ def run_test(self):
         self.nodes[0].setlabel(change, 'foobar')
         assert_equal(self.nodes[0].getaddressinfo(change)['ischange'], False)
 
+        # Test gettransaction for showing CoinJoin transaction. See issue #14136.
+        # Build a tx where inputs = [25 mine, 25 not-mine] and outputs = [49 mine, 0.99 not-mine].
+        # The amount returned by gettransaction is expected to be 49 - 25 = 24.
+        self.nodes[0].generate(1)
+        input_address_0 = self.nodes[0].getnewaddress()
+        input_address_1 = self.nodes[1].getnewaddress()
+        input_privkey_0 = self.nodes[0].dumpprivkey(input_address_0)
+        input_privkey_1 = self.nodes[1].dumpprivkey(input_address_1)
+        output_address_0 = self.nodes[0].getnewaddress()
+        output_address_1 = self.nodes[1].getnewaddress()
+        txid_0 = self.nodes[0].getblock(self.nodes[0].generatetoaddress(1, input_address_0)[0])['tx'][0]
```

Comment:
At this point `self.nodes[1]` has some UTXO that can be used instead of generating more blocks.

Reply:
Obtaining txids and amounts of those existing UTXOs would make the code much more complex. Is it necessary?

Reply:
Just `listunspent`?